swim/test_utils.go

```diff
@@ -121,6 +123,52 @@ func bootstrapNodes(t *testing.T, testNodes ...*testNode) {
 		})
 		require.NoError(t, err, "node must bootstrap successfully")
 	}
+	return hostports
+}
+
+func waitForConvergence(t *testing.T, timeout time.Duration, testNodes ...*testNode) {
+	var wg sync.WaitGroup
+	for _, tn := range testNodes {
+		wg.Add(1)
+		// execute the Protocol period on all nodes until the dissemination list is exhausted
+		go func(tn *testNode) {
+			for len(tn.node.disseminator.changes) > 0 {
```

Comment:
Exhausting changes seems like a useful API to have, if not at the `swim.Node`, then definitely the test suite API layer. I recommend that we pull out the `swimCluster` type from test_utils and add `ExhaustChanges()` to it.